test/main.jl

```diff
@@ -0,0 +1,127 @@
+using Test
+import IJulia
+
+@testset "@main app entry point" begin
+    # Test help output
+    @testset "help" begin
+        # Capture stdout using a pipe
+        old_stdout = stdout
+        rd, wr = redirect_stdout()
+
+        ret = IJulia.main(["--help"])
+
+        # Restore stdout and read the captured output
+        redirect_stdout(old_stdout)
+        close(wr)
+        help_text = read(rd, String)
+        close(rd)
+
+        @test ret == 0
+        @test occursin("IJulia Launcher", help_text)
+        @test occursin("notebook", help_text)
+        @test occursin("lab", help_text)
+        @test occursin("--dir=PATH", help_text)
+        @test occursin("--port=N", help_text)
+        @test occursin("--detached", help_text)
+        @test occursin("--verbose", help_text)
+    end
+
+    # Test invalid subcommand
+    @testset "invalid subcommand" begin
+        ret = @test_logs (:error, r"Unknown subcommand.*Use 'notebook' or 'lab'") IJulia.main(["invalid"])
+        @test ret == 1
+    end
+
+    # Test argument parsing without actually launching
+    # We use mutable function references (notebook_cmd/jupyterlab_cmd) to test parsing
+    @testset "argument parsing" begin
+        # Save original functions
+        orig_notebook = IJulia.notebook_cmd
+        orig_jupyterlab = IJulia.jupyterlab_cmd
```

Comment:
Nitpicking, I'd move these to the toplevel testset so each sub-testset doesn't need to store them.